src/data/auroradb.ts

```diff
@@ -0,0 +1,249 @@
+    street: string;
+}
+
+interface NaptanAtcoCodes {
+    naptanCode: string;
+    atcoCode: string;
+}
+
+export interface Stop {
+    stopName: string;
+    naptanCode: string;
+    atcoCode: string;
+    localityCode: string;
+    localityName: string;
+    parentLocalityName: string;
+    indicator?: string;
+    street?: string;
+    qualifierName?: string;
+}
+
+export interface StopIdentifiers {
+    naptanCode: string | null;
+    atcoCode: string;
+}
+
+export interface Service {
+    serviceDescription: string;
+    operatorShortName: string;
+    journeyPatterns: JourneyPattern[];
+}
+
+export interface RawService {
+    serviceDescription: string;
+    operatorShortName: string;
+    journeyPatterns: RawJourneyPattern[];
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+const getAuroraDBClient = () => {
+    let client = null;
+
+    if (process.env.NODE_ENV === 'development') {
+        client = createConnection({
+            host: 'mysql',
+            user: 'fdbt_site',
+            password: 'password',
+            database: 'fdbt',
+        });
+    } else {
+        // TODO find replacement here
+        client = createConnection({
+            host: 'mysql',
+            user: 'fdbt_site',
+            password: 'password',
+            database: 'fdbt',
+        });
+    }
+
+    return client;
+};
+
+export const convertDateFormat = (startDate: string): string => {
+    return dateFormat(startDate, 'dd/mm/yyyy');
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const executeQuery = async (query: string, values: any[]) => {
+    const connection = await getAuroraDBClient();
+    const [rows] = await connection.execute(query, values);
+    connection.end();
+    return rows;
+};
+
+export const getServicesByNocCode = async (nocCode: string): Promise<ServiceType[]> => {
+    try {
+        const queryInput = 'SELECT * FROM `tndsService` WHERE `nocCode` = ?';
+
+        const queryResult = await executeQuery(queryInput, [nocCode]);
+
+        const resultRow = JSON.parse(JSON.stringify(queryResult));
+
+        return (
+            resultRow?.map((item: ServiceType) => ({
+                lineName: item.lineName,
+                startDate: convertDateFormat(item.startDate),
+                description: item.description,
+            })) || []
+        );
+    } catch (err) {
+        throw new Error(`Could not retrieve services from DynamoDB: ${err.name}, ${err.message}`);
+    }
+};
+
+export const batchGetStopsByAtcoCode = async (atcoCodes: string[]): Promise<Stop[] | []> => {
+    let orQuery = '';
+
+    try {
+        atcoCodes.forEach((code, index) => {
+            orQuery += `atcoCode = ${code} ${index < atcoCodes.length - 1 ? 'OR ' : ''}`;
+        });
+
+        const batchQuery = `Select * from naptanStop where ${orQuery}`;
+
+        const results = await executeQuery(batchQuery, []);
+
+        const parsedResults = JSON.parse(JSON.stringify(results));
```

Comment:
Have to stringify then parse it which is not logical. but the results we get back in another format.

Reply:
That's odd, what's the difference before and after?

Reply:
currently result comes back like this

```
[
  BinaryRow {
    atcoCode: '0100053306',
    naptanCode: 'bstapjm',
```

if you just parse on the result above then you get

> Error performing batch get for naptan info for stop list: 0100053264, error: SyntaxError: Unexpected token o in JSON at position 1
> Error: SyntaxError: Unexpected token o in JSON at position 1